src/CmsManager/BaseCmsDashboardManager.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DashboardBundle\CmsManager;
+
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\DashboardBundle\Model\DashboardInterface;
+
+/**
+ * Base class CMS Manager.
+ *
+ * @author Stephane PY <py.stephane1@gmail.com>
+ */
+abstract class BaseCmsDashboardManager implements CmsManagerInterface
+{
+    /**
+     * @var DashboardInterface|null
+     */
+    protected $currentDashboard;
+
+    /**
+     * @var BlockInterface[]
+     */
+    protected $blocks = [];
```

Comment:
Those should be private since they have accessors… wait, how do you add a new block?

Reply:
This is an abstract class. We need this in the implementations